chain/chain/src/store.rs

```diff
@@ -2320,6 +2320,42 @@ impl<'a> ChainStoreUpdate<'a> {
         shard_uids_to_gc
     }
 
+    /// GC trie state and flat state data after a resharding event
+    /// Most of the work happens on the last block of the epoch when resharding is COMPLETED
+    /// During GC, when we detect a change in shard layout, we can clear off all entries from
+    /// the parent shards
+    pub fn clear_resharding_data(
+        &mut self,
+        runtime: &dyn RuntimeAdapter,
+        epoch_manager: &dyn EpochManagerAdapter,
+        block_hash: CryptoHash,
+    ) -> Result<(), Error> {
+        // Need to check if this is the last block of the epoch where resharding is completed
+        // which means shard layout changed in the previous epoch
```

Comment:
Just for my complete understanding: we are cleaning up the last block where the parent shard layout was used so we can clean up the relevant state and flat state as well. Is that statement correct?

Reply:
Yep, that's correct.
Epoch T - 1 : Old shard layout being used
Epoch T : Resharding begins, we are still mainly using old shard layout but new shards are being built and updated
Epoch T + 1 : We switch to new shard layout.

We are doing the cleanup at the last block of Epoch T.